### PR TITLE
Migrate Zigbee unit tests to new ZTest API [KRKNWK 16665]

### PR DESCRIPTION
**Pull request by sebastiandraus**

### CODE REVIEW

File: tests/subsys/zigbee/osif/nvram/src/main.c [hunk 1]
```diff
@@ -65,7 +67,7 @@ static void test_zb_nvram_erase(void)
 	}
 }
 
-static void test_zb_nvram_write(void)
+ZTEST(osif_test, test_zb_nvram_write)
 {
```
Comment (tomchy) [inline tests/subsys/zigbee/osif/nvram/src/main.c:71]: (un)fortunately the new ZTEST API does not allow to rely on the order of tests (they are executed alphabetically or in a random order), so this test requires a small setUp function that will erase the area before writing to it.

**Review by gopiotr — CHANGES_REQUESTED** [review]
Please remove in this PR (in separate commit) Zigbee tests from quarantine file:
https://github.com/nrfconnect/sdk-nrf/blob/57cccdab05311a13aaf6d8c329913b3fff27fda1/scripts/quarantine.yaml#L23-L28
Thanks to this we would be able to build zigbee tests in CI dedicated for this PR

**Review by mariuszpos — APPROVED** [review]

File: tests/subsys/zigbee/osif/nvram/src/main.c (hunk 1 above)
Comment (sebastiandraus) [reply tests/subsys/zigbee/osif/nvram/src/main.c:83]: Good catch, fix already applied :)

**Comment by sebastiandraus** [comment]
> Please remove in this PR (in separate commit) Zigbee tests from quarantine file:
> 
> https://github.com/nrfconnect/sdk-nrf/blob/57cccdab05311a13aaf6d8c329913b3fff27fda1/scripts/quarantine.yaml#L23-L28
> 
> Thanks to this we would be able at least build zigbee tests in CI dedicated for this PR

Done!

File: tests/subsys/zigbee/osif/serial/serial_async_api/src/main.c [hunk 2]
```diff
@@ -324,48 +331,6 @@ void test_main(void)
 	/* Enable zigbee callbacks implemented in zigbee_scheduler mock. */
 	zigbee_enable();
```
Comment (gopiotr) [inline tests/subsys/zigbee/osif/serial/serial_async_api/src/main.c:332]: Is there anything against moving this `zigbee_enable()` and those `printk` to testsuite setup function `configure_test_suite`? Thanks to this you could remove `test_main` function from this test.

File: tests/subsys/zigbee/osif/serial/serial_basic_api/src/main.c [hunk 3]
```diff
@@ -167,24 +173,6 @@ void test_main(void)
 	/* Enable zigbee callbacks implemented in zigbee_scheduler mock. */
 	zigbee_enable();
```
Comment (gopiotr) [inline tests/subsys/zigbee/osif/serial/serial_basic_api/src/main.c:174]: The same issue as in `serial_basic_api`.

**Review by gopiotr — CHANGES_REQUESTED** [review]

File: tests/subsys/zigbee/osif/serial/serial_async_api/src/main.c (hunk 2 above)
Comment (sebastiandraus) [reply tests/subsys/zigbee/osif/serial/serial_async_api/src/main.c]: Good idea, already applied :)

**Review by gopiotr — APPROVED** [review]

**Review by tomchy — APPROVED** [review]

**Pull request merged**